scripts/scripts.js

```diff
@@ -783,6 +783,87 @@ const preflightListener = async () => {
   customModal.showModal();
 };
 
+// Set event for the publish button for confirmation message
+const publishConfirmationPopUp = (oSidekick, oPublishButtons) => {
+  // Add plugin listeners here
+  if (!oSidekick) {
+    return;
+  }
+  oSidekick.addEventListener('custom:preflight', preflightListener);
+  oPublishButtons.forEach((oPublishBtn) => {
+    // eslint-disable-next-line func-names, consistent-return
+    oPublishBtn.addEventListener('mousedown', function (e) {
+      // eslint-disable-next-line no-restricted-globals, no-alert
+      if (confirm('Are you sure you want to publish this content live?')) {
+        // continue publishing
+        this.click();
+      } else {
+        // avoid publishing
+        e.stopImmediatePropagation();
+        return false;
+      }
+    });
+  });
+};
+
+// Handler for publish button to set observer-
+// if the publish button is already loaded on the shadowroot
+const publishConfirmationHandler = (oSidekick) => {
+  if (!oSidekick) {
+    return;
+  }
+  const oShadowRoot = oSidekick.shadowRoot;
+  if (!oShadowRoot) {
+    return;
+  }
+
+  // Options for the observer (which mutations to observe)
+  const config = { childList: true, subtree: true };
+
+  // Callback function to execute when mutations are observed
+  const callback = (_mutationList, observer) => {
+    const oPublishButtons = oSidekick.shadowRoot.querySelectorAll('button[title="Publish"]');
+    if (oPublishButtons.length !== 0) {
+      publishConfirmationPopUp(oSidekick, oPublishButtons);
+      observer.disconnect();
+    }
+  };
+
+  // Create an observer instance linked to the callback function
+  const observer = new MutationObserver(callback);
+
+  // Start observing the target node for configured mutations
+  observer.observe(oShadowRoot, config);
+};
+
+// Observe helix-sidekick element if already loaded on the html body
+const helixSideKickObserver = () => {
+  const oSidekick = document.querySelector('helix-sidekick');
+  if (oSidekick) {
+    publishConfirmationHandler(oSidekick);
+    return;
+  }
+  const oBody = document.querySelector('body');
+
+  // Options for the observer (which mutations to observe)
+  const config = { childList: true };
+
+  // Callback function to execute when mutations are observed
+  const callback = (_mutationList, observer) => {
+    const oAddedSidekick = document.querySelector('helix-sidekick');
```

Comment:
@jjjaspher rather than using the mutation observers, did you try using the `sidekick-ready` event like this example 
```
const sk = document.querySelector('helix-sidekick');
if (sk) {
  // sidekick already loaded
  sk.addEventListener('custom:foo', foo);
} else {
  // wait for sidekick to be loaded
  document.addEventListener('sidekick-ready', () => {
    document.querySelector('helix-sidekick')
      .addEventListener('custom:foo', foo);
  }, { once: true });
}
```
in https://www.aem.live/developer/sidekick-customization#plugins?

cc: @raymond-h-dev

Reply:
thanks for the reference @rpapani , @jjjaspher could we try Ravi's suggestions?

Reply:
Hi @ravkiran @raymond-h-dev, we can used the `sidekick-ready` event only for checking the helix-sidekick element but I think we still need the observer for publish button. Upon checking loading of the publish button has a delay.
![image](https://github.com/hlxsites/accenture-newsroom/assets/142872664/11f66582-9c78-4bce-be21-d76776d68f8b)
![image](https://github.com/hlxsites/accenture-newsroom/assets/142872664/73d12829-2bc7-43fc-9781-8190323e0e05)